contracts/Migrator.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import "OpenZeppelin/openzeppelin-contracts@4.6.0/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import "./Governable.sol";
+
+interface IStaking {
+    function delegates(address staker) external view returns (address);
+
+    // From OGVStaking.sol
+    function unstakeFrom(address staker, uint256[] memory lockupIds) external returns (uint256, uint256);
+
+    // From ExponentialStaking.sol
+    function stake(uint256 amountIn, uint256 duration, address to, bool stakeRewards, int256 lockupId) external;
+}
+
+contract Migrator is Governable {
+    ERC20Burnable public immutable ogv;
+    ERC20Burnable public immutable ogn;
+
+    IStaking public immutable ogvStaking;
+    IStaking public immutable ognStaking;
+
+    // Fixed conversion rate
+    uint256 public constant CONVERSION_RATE = 0.09137 ether;
+
+    uint256 public endTime;
+
+    event TokenExchanged(uint256 ogvAmountIn, uint256 ognAmountOut);
+    event Decommissioned();
+    event LockupsMigrated(address indexed user, uint256[] ogvLockupIds, uint256 newDuration);
+
+    error MigrationAlreadyStarted();
+    error MigrationIsInactive();
+    error MigrationNotComplete();
+    error BalanceMismatch(uint256 ogvAmountIn, uint256 expectedOgnAmountOut, uint256 actualOgnAmountOut);
+    error LockupIdsRequired();
+
+    constructor(address _ogv, address _ogn, address _ogvStaking, address _ognStaking) {
+        ogv = ERC20Burnable(_ogv);
+        ogn = ERC20Burnable(_ogn);
+        ogvStaking = IStaking(_ogvStaking);
+        ognStaking = IStaking(_ognStaking);
+    }
+
+    /**
+     * @notice Starts the migration and sets it to end after
+     *          365 days. Also, approves xOGN to transfer OGN
+     *          held in this contract. Can be invoked only once
+     */
+    function start() external onlyGovernor {
+        if (endTime != 0) {
+            revert MigrationAlreadyStarted();
+        }
+
+        // Max approve
+        ogn.approve(address(ognStaking), type(uint256).max);
+
+        endTime = block.timestamp + 365 days;
+    }
```

Comment:
Let's do our isSolvent check here as well, making sure that as soon as this is called, the migration is good to go.

Reply:
Have added that here and for `transferExcessTokens()` method too